test: 대시보드 UI playwright 구조 검증 테스트 추가

프론트엔드 수정 시 UI 깨짐을 코드 레벨에서 탐지할 수 없는 문제. playwright로 서버 띄우고 주요 요소 존재 여부 + 네비게이션 동작 + 뷰 전환 테스트 추가. 이후 HTML/JS 분리 작업의 안전망 역할.

## AI Quartermaster - 파이프라인 실패

자동 구현에 실패했습니다.

**에러**: Cannot find module '/home/by/.ai-quartermaster/src/pipeline/checkpoint.js' imported from /home/by/.ai-quartermaster/src/cli.ts

수동 확인이 필요합니다.

## AI Quartermaster - 파이프라인 실패

자동 구현에 실패했습니다.

**에러**: Cannot find module '/home/by/.ai-quartermaster/src/pipeline/checkpoint.js' imported from /home/by/.ai-quartermaster/src/cli.ts

수동 확인이 필요합니다.

## AI Quartermaster - 파이프라인 실패

자동 구현에 실패했습니다.

**에러**: Cannot find module '/home/by/.ai-quartermaster/src/pipeline/checkpoint.js' imported from /home/by/.ai-quartermaster/src/cli.ts

수동 확인이 필요합니다.

--- a/package.json
+++ b/package.json
@@ -23,6 +23,7 @@
     "postbuild": "node scripts/add-shebang.cjs",
     "prepublishOnly": "npm run typecheck && npm run lint && npm run test && npm run build",
     "test": "vitest run",
+    "test:e2e": "playwright test",
     "test:watch": "vitest",
     "lint": "eslint src/ tests/",
     "typecheck": "tsc --noEmit"
@@ -39,6 +40,7 @@
     "zod": "^3.22.0"
   },
   "devDependencies": {
+    "@playwright/test": "^1.59.1",
     "@types/better-sqlite3": "^7.6.13",
     "@types/node": "^20.11.0",
     "@types/node-cron": "^3.0.11",

--- a/playwright.config.ts
+++ b/playwright.config.ts
@@ -1,0 +1,26 @@
+import { defineConfig, devices } from '@playwright/test';
+
+export default defineConfig({
+  testDir: './tests/e2e',
+  fullyParallel: false,
+  forbidOnly: !!process.env['CI'],
+  retries: process.env['CI'] ? 1 : 0,
+  workers: 1,
+  reporter: 'list',
+  use: {
+    baseURL: 'http://localhost:3100',
+    trace: 'on-first-retry',
+  },
+  projects: [
+    {
+      name: 'chromium',
+      use: { ...devices['Desktop Chrome'] },
+    },
+  ],
+  webServer: {
+    command: 'tsx src/cli.ts start --port 3100',
+    url: 'http://localhost:3100/health',
+    reuseExistingServer: !process.env['CI'],
+    timeout: 30000,
+  },
+});

--- a/tests/e2e/ui/dashboard-navigation.test.ts
+++ b/tests/e2e/ui/dashboard-navigation.test.ts
@@ -1,0 +1,90 @@
+import { test, expect } from '@playwright/test';
+
+const VIEWS = ['dashboard', 'logs', 'repositories', 'automations', 'settings'] as const;
+type ViewName = typeof VIEWS[number];
+
+// Header nav only exposes dashboard and logs
+const HEADER_NAV_VIEWS: ViewName[] = ['dashboard', 'logs'];
+
+test.describe('Dashboard Navigation', () => {
+  test.beforeEach(async ({ page }) => {
+    await page.goto('/');
+  });
+
+  test('초기 상태: dashboard 뷰가 active', async ({ page }) => {
+    await expect(page.locator('#view-dashboard')).toHaveClass(/active/);
+
+    for (const view of VIEWS.filter((v) => v !== 'dashboard')) {
+      await expect(page.locator(`#view-${view}`)).not.toHaveClass(/active/);
+    }
+
+    await expect(page.locator('#sidebar-nav a[data-nav="dashboard"]')).toHaveClass(/nav-item-active/);
+  });
+
+  // ── Sidebar navigation ──────────────────────────────────────────────────────
+
+  for (const view of VIEWS) {
+    test(`사이드바 → ${view}: 해당 뷰 활성화`, async ({ page }) => {
+      await page.locator(`#sidebar-nav a[data-nav="${view}"]`).click();
+
+      // Target view gains active class
+      await expect(page.locator(`#view-${view}`)).toHaveClass(/active/);
+
+      // All other views lose active class
+      for (const other of VIEWS.filter((v) => v !== view)) {
+        await expect(page.locator(`#view-${other}`)).not.toHaveClass(/active/);
+      }
+
+      // Clicked sidebar item gains nav-item-active
+      await expect(page.locator(`#sidebar-nav a[data-nav="${view}"]`)).toHaveClass(/nav-item-active/);
+
+      // All other sidebar items lose nav-item-active
+      for (const other of VIEWS.filter((v) => v !== view)) {
+        await expect(page.locator(`#sidebar-nav a[data-nav="${other}"]`)).not.toHaveClass(/nav-item-active/);
+      }
+    });
+  }
+
+  // ── Header navigation ───────────────────────────────────────────────────────
+
+  for (const view of HEADER_NAV_VIEWS) {
+    test(`헤더 nav → ${view}: 해당 뷰 활성화`, async ({ page }) => {
+      // Navigate away first so we have a meaningful transition to test
+      const startView: ViewName = view === 'dashboard' ? 'logs' : 'dashboard';
+      await page.locator(`#sidebar-nav a[data-nav="${startView}"]`).click();
+      await expect(page.locator(`#view-${startView}`)).toHaveClass(/active/);
+
+      // Click header nav link
+      await page.locator(`header nav a[data-nav="${view}"]`).click();
+
+      // Target view becomes active
+      await expect(page.locator(`#view-${view}`)).toHaveClass(/active/);
+
+      // Previous view becomes inactive
+      await expect(page.locator(`#view-${startView}`)).not.toHaveClass(/active/);
+
+      // Active header link has border-b-2
+      await expect(page.locator(`header nav a[data-nav="${view}"]`)).toHaveClass(/border-b-2/);
+
+      // Inactive header links do not have border-b-2
+      for (const other of HEADER_NAV_VIEWS.filter((v) => v !== view)) {
+        await expect(page.locator(`header nav a[data-nav="${other}"]`)).not.toHaveClass(/border-b-2/);
+      }
+    });
+  }
+
+  // ── Full cycle traversal ─────────────────────────────────────────────────────
+
+  test('5개 뷰 순회: 사이드바로 모든 뷰 전환 검증', async ({ page }) => {
+    for (const view of VIEWS) {
+      await page.locator(`#sidebar-nav a[data-nav="${view}"]`).click();
+
+      await expect(page.locator(`#view-${view}`)).toHaveClass(/active/);
+      await expect(page.locator(`#sidebar-nav a[data-nav="${view}"]`)).toHaveClass(/nav-item-active/);
+
+      // Exactly one view-panel should be active at any time
+      const activePanels = page.locator('.view-panel.active');
+      await expect(activePanels).toHaveCount(1);
+    }
+  });
+});

--- a/tests/e2e/ui/dashboard-structure.test.ts
+++ b/tests/e2e/ui/dashboard-structure.test.ts
@@ -1,0 +1,43 @@
+import { test, expect } from '@playwright/test';
+
+test.describe('대시보드 레이아웃 구조 검증', () => {
+  test.beforeEach(async ({ page }) => {
+    await page.goto('/');
+  });
+
+  test('사이드바 존재 및 5개 nav 항목 확인', async ({ page }) => {
+    const sidebarNav = page.locator('#sidebar-nav');
+    await expect(sidebarNav).toBeVisible();
+
+    const navKeys = ['dashboard', 'logs', 'repositories', 'automations', 'settings'];
+    for (const nav of navKeys) {
+      await expect(page.locator(`#sidebar-nav [data-nav="${nav}"]`)).toBeVisible();
+    }
+  });
+
+  test('헤더 바 + 프로젝트 셀렉터 + 테마 토글 + 언어 토글 + 접속 상태 표시 확인', async ({ page }) => {
+    await expect(page.locator('header')).toBeVisible();
+    await expect(page.locator('#project-selector')).toBeVisible();
+    await expect(page.locator('#btn-theme')).toBeVisible();
+    await expect(page.locator('#lang-label')).toBeVisible();
+    await expect(page.locator('#conn-dot')).toBeAttached();
+    await expect(page.locator('#conn-label')).toBeAttached();
+  });
+
+  test('기본 뷰(dashboard)가 active 상태', async ({ page }) => {
+    await expect(page.locator('#view-dashboard')).toHaveClass(/active/);
+  });
+
+  test('5개 view-panel 요소가 DOM에 존재', async ({ page }) => {
+    const viewIds = [
+      'view-dashboard',
+      'view-logs',
+      'view-repositories',
+      'view-automations',
+      'view-settings',
+    ];
+    for (const id of viewIds) {
+      await expect(page.locator(`#${id}`)).toBeAttached();
+    }
+  });
+});

--- a/vitest.config.ts
+++ b/vitest.config.ts
@@ -7,6 +7,7 @@ export default defineConfig({
     },
   },
   test: {
+    exclude: ['node_modules', 'dist', 'tests/e2e/**'],
     coverage: {
       provider: 'v8',
       reporter: ['text', 'json-summary'],

